markets/perps-market/package.json

```diff
@@ -4,28 +4,36 @@
   "description": "Spot Market implementation",
   "private": true,
   "scripts": {
-    "build": "hardhat storage:verify && hardhat cannon:build",
-    "start": "npm run build && npm run cannon",
+    "build1": "hardhat storage:verify && hardhat cannon:build",
+    "check:storage": "git diff --exit-code storage.dump.sol",
+    "start": "yarn build && yarn cannon-build",
     "clean": "hardhat clean",
-    "cannon": "npx cannon perps-market:0.0.3-alpha",
+    "cannon-build": "cannon perps-market:0.0.3-alpha",
     "test": "hardhat test",
-    "coverage": "npm run test # npm run build && hardhat coverage --network hardhat",
+    "coverage": "yarn test",
     "compile-contracts": "hardhat compile",
     "size-contracts": "hardhat compile && hardhat size-contracts"
   },
   "keywords": [],
   "author": "Synthetix",
   "license": "MIT",
   "devDependencies": {
-    "@synthetixio/common-config": "*",
-    "@synthetixio/core-contracts": "1.1.1",
-    "@synthetixio/core-modules": "*",
-    "@synthetixio/main": "*",
-    "@synthetixio/router": "*",
-    "node-fetch": "2.6.9",
-    "solidity-docgen": "0.6.0-beta.22",
+    "@ethersproject/abi": "5.7.0",
+    "@ethersproject/providers": "5.7.2",
+    "@nomiclabs/hardhat-ethers": "2.2.1",
+    "@synthetixio/common-config": "workspace:*",
+    "@synthetixio/core-contracts": "workspace:*",
+    "@synthetixio/core-modules": "workspace:*",
+    "@synthetixio/main": "workspace:*",
```

Comment:
so this probably isnt relevant to this specific PR, but the plan here was that cannon would read the version of `synthetix:<version>` to depend on from this variable. In order to do this, the version would be `@synthetixio/main: 3.0.1` for example or whatever the version is, with no `workspace:` or any other addition. Then, cannonfile imports the correct version with `source = "synthetix:<%= package.devDependencies['@synthetixio/main'] %>"` (as an example). Finally, lerna automatically bumps the version any time the dependency is updated, and cannon remains happy. guess what I am saying is, LMK if you have an alternative plan at this point.

Reply:
we have a lot of these types of dependencies within the project. They are basically dependencies only insofar as that we are using the solidity sources and cannon package matching, so it makes sense to consider that these packages may have their version resolved differently.

Reply:
oh one more thing, wouldn't `workspace:*` resolve to `*` for the versions when the pacakge is actually published? Is this intended behavior for our internal javascript packages? its not a big deal for us, but we might as well try to do things reasonably when possible.